File: dataStructures&algorithms/algorithm/leetCode/aboutArray/mergeSortedArr/mergeSortedArr.php
<?php
/**
 * @date: 2022/6/1
 * @createTime: 8:05 PM
 */

// leetcode-88
function mergeSortedArr($arr1, $arr2)
{
    $len1 = count($arr1);
    $len2 = count($arr2);
    $i = $j = 0;
    $res = [];
    while(1) {
        if($i == $len1) {
            array_push($res, ...array_slice($arr2, $j));
            break;
        }
        if($j == $len2) {
            array_push($res, ...array_slice($arr1, $i));
            break;
        }
        if($arr1[$i] < $arr2[$j]) {
            $res[] = $arr1[$i];
            $i++;
        } else {
            $res[] = $arr2[$j];
            $j++;
        }
    }
    return $res;
}

$arr1 = [1,2,3];
$arr2 = [2,5,6];
print_r(mergeSortedArr($arr1, $arr2));

File: dataStructures&algorithms/algorithm/leetCode/aboutArray/mergeSortedArr/mergeSortedArr.go
/**
 * @Author: shershon
 * @Description:
 * @Date: 2023/03/06 15:11
 */

package mergeSortedArr

// leetcode-88
func mergeSortedArr(a, b []int) (result []int) {
	len1 := len(a)
	len2 := len(b)
	var i, j int = 0, 0
	for {
		if i == len1 {
			result = append(result, b[i-1:]...)
			break
		}
		if j == len2 {
			result = append(result, a[j-1:]...)
			break
		}
		if a[i] < b[j] {
			result = append(result, a[i])
			i++
		} else {
			result = append(result, b[j])
			j++
		}
	}
	return result
}
